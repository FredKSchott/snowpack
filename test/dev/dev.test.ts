const path = require('path');
const Stream = require('stream');

const execa = require('execa');
const {readdirSync, readFileSync, statSync, existsSync} = require('fs');
const glob = require('glob');
const os = require('os');
const got = require('got');

describe('snowpack dev', () => {
  it('smoke', async () => {
    const cwd = path.join(__dirname, 'smoke');
    const streamLogsToOutput = new Stream.Writable({objectMode: true});

    // start the server
    const snowpackProcess = execa('yarn', ['teststart'], {cwd});
    snowpackProcess.stdout.pipe(streamLogsToOutput);

    // await server to be ready and set a timeout in case something goes wrong
    await new Promise((resolve, reject) => {
      // start timeout in case something goes wrong.
      const timeout = setTimeout(() => {
        snowpackProcess.cancel();
        console.error(output.join(''));
        reject(new Error('Timeout: snowpack did not start server within 3 seconds.'));
      }, 3000);

      const output = [];
      streamLogsToOutput._write = (buffer, encoding, done) => {
        const line = buffer.toString();
        output.push(line);
        if (/Server started in/.test(line)) {
          resolve();
          clearTimeout(timeout);
        }

        done();
      };
    });

    // get HTML
    const {body: htmlBody} = await got('http://localhost:8080');
    expect(htmlBody).toMatchSnapshot('html');

    // get built JS
    const {body: jsBody} = await got('http://localhost:8080/_dist_/index.js');
    expect(jsBody).toMatchSnapshot('js');

    // all set
    snowpackProcess.cancel();
  });
});
