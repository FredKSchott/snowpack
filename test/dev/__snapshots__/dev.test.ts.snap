// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snowpack dev smoke: html 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <meta charset=\\"utf-8\\" />
    <link rel=\\"icon\\" href=\\"/favicon.ico\\" />
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\" />
    <meta name=\\"description\\" content=\\"Web site created using create-snowpack-app\\" />
    <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/index.css\\" />
    <title>Snowpack App</title>
  <script type=\\"module\\" src=\\"/__snowpack__/hmr-client.js\\"></script><script type=\\"module\\" src=\\"/__snowpack__/hmr-error-overlay.js\\"></script></head>
  <body>
    <img id=\\"img\\" src=\\"/logo.svg\\" />
    <canvas id=\\"canvas\\"></canvas>
    <noscript>You need to enable JavaScript to run this app.</noscript>
    <script type=\\"module\\" src=\\"/_dist_/index.js\\"></script>
    <!--
      This HTML file is a template.
      If you open it directly in the browser, you will see an empty page.

      You can add webfonts, meta tags, or analytics to this file.
      The build step will place the bundled scripts into the <body> tag.

      To begin the development, run \`npm start\` or \`yarn start\`.
      To create a production bundle, use \`npm run build\` or \`yarn build\`.
    -->
  </body>
</html>
"
`;

exports[`snowpack dev smoke: js 1`] = `
"/**
 * This file is just a silly example to show everything working in the browser.
 * When you're ready to start on your site, clear the file. Happy hacking!
 **/

import confetti from '/web_modules/canvas-confetti.js';

confetti.create(document.getElementById('canvas'), {
  resize: true,
  useWorker: true,
})({ particleCount: 200, spread: 200 });
"
`;
